snippet_parser/core.py

```diff
@@ -169,6 +169,11 @@ def extract(self, wikitext):
                     marked = _CITATION_NEEDED_MARKER_MARKUP.format(tpl = tpl)
                     try:
                         section.replace(tpl, marked)
+                        try:
+                            self.date = tpl.get('date')
```

Comment:
Hmm, I think we'll need a bit more cleanup first, sorry. We don't want to just save the date to `self.date`.

This method will typically extract multiple snippets from an article, and each snippet will have its own `{{cn}}` template (or even more than one).  So each snippet will have its own date and we need to return all snippets and all dates to the caller of this method.

Currently, this method returns an awkward structure, documented above in the docstring:

```python
            [
                [<section1>, [<snippet1>, <snippet2>, ...]],
                [<section2>, [<snippet1>, ...]],
                ...
            ]
```

How about we create a simple Snippet class like so:

```python
class Snippet(object):
  def __init__(self, html, section, date):
    self.html = html
    self.section = section
    self.date = date
```

and return something more sensible like a dict:

```python
  {<section1>: [Snippet1, Snippet2], <section2>: [Snippet3, Snippet4]}
```

? That way we can easily group together the snippets and dates.

Basically [here](https://github.com/eggpi/citationhunt/blob/master/snippet_parser/core.py#L279) you would instead do something like:

```python
  sections.extend(Snippet(snippet, sectitle, '') for snippet in snippets_in_section)
```

For now I'm passing a dummy `''` as the `date` because we still need to figure out how to propagate the date of each snippet all the way down there. I think I know how to do it, but let's do this cleanup first please...